tests/zfs-tests/tests/functional/cli_root/zpool_set/vdev_set_queue_io.ksh

```diff
@@ -0,0 +1,96 @@
+#!/bin/ksh -p
+# SPDX-License-Identifier: CDDL-1.0
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or https://opensource.org/licenses/CDDL-1.0.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright (c) 2024 by Triad National Security, LLC.
+#
+
+. $STF_SUITE/include/libtest.shlib
+
+#
+# DESCRIPTION:
+# Toggling vdev queue_io property while reading from vdev should not cause panic.
+#
+# STRATEGY:
+# 1. Create a zpool
+# 2. Write a file to the pool.
+# 3. Start reading from file, while also toggling the queue_io property on / off.
+#
+
+verify_runnable "global"
+
+command -v fio > /dev/null || log_unsupported "fio missing"
+log_must save_tunable DIO_ENABLED
+log_must set_tunable32 DIO_ENABLED 1
+
+function toggle_queue_io
+{
+	zpool set queue_io=off $TESTPOOL1 $FILEDEV
+	sleep 0.1
+	zpool set queue_io=on $TESTPOOL1 $FILEDEV
+	sleep 0.1
+}
+
+function cleanup
+{
+	log_must destroy_pool $TESTPOOL1
+	rm -f $FILEDEV
+	log_must restore_tunable DIO_ENABLED
+}
+
+log_assert "Toggling vdev queue_io property while reading from vdev should not cause panic"
+log_onexit cleanup
+
+# 1. Create a pool
+
+FILEDEV="$TEST_BASE_DIR/filedev.$$"
+log_must truncate -s $(($MINVDEVSIZE * 2)) $FILEDEV
+log_must create_pool $TESTPOOL1 $FILEDEV
+
+mntpnt=$(get_prop mountpoint $TESTPOOL1)
+
+# 2. Write a file to the pool, while also toggling the queue_io property on / off.
+
+log_must eval "fio --filename=$mntpnt/foobar --name=write-file \
+		--rw=write --size=$MINVDEVSIZE --bs=128k --numjobs=1 --direct=1 \
+		--ioengine=sync --runtime=10 &"
```

Comment:
Don't we need to add `--time_based` like below to ensure `fio` will loop and run for the full 10 seconds.  From `fio(1)`.

```
       runtime=time
              Tell fio to terminate processing after the specified period of time. It can be quite
              hard to determine for how long a specified job will run, so this parameter is  handy
              to  cap  the  total  runtime to a given time. When the unit is omitted, the value is
              interpreted in seconds.

       time_based
              If set, fio will run for the duration of the runtime specified even if  the  file(s)
              are  completely  read or written. It will simply loop over the same workload as many
              times as the runtime allows.
```